### PR TITLE
feat : edit and delete datasource functionality 

### PR DESCRIPTION
**Pull request by ManishChain**
There is a DELTE button for each datasource and popup a confirmation dialog upon action

As discussed with Vincent, we can implement the delete function at this stage; together with other tasks; and keep the “edit” for later version. view JIRA issue https://settlemint.atlassian.net/browse/DDX-6

### CODE REVIEW

**Comment by ManishChain** [comment]
@woodyjon Can you help me identifying why tests are failing ?

**Comment by ManishChain** [comment]
@woodyjon Could you please update on this PR ? If you can help me on this failing checks then it would be great. I am not able to resolve these.

**Comment by ManishChain** [comment]
@goosecoid  Steps
1. Clone branch
2. Optionally configure .env 
3. cd dxc <root folder>
4. Issue command: go run server.go 
5. Another terminal: cd <root>/ui 
6. Issue command: npm run start
7. localhost:3000 started in browser

DELETE feature
1. Add few datasources 
2. Each datasource have DELETE button. On click of it, confirm dialog appears. If you click OK then datasource is deleted  

EDIT
1. Add few datasources
2. Each datasource have DELETE button. On click of it, confirm dialog appears. 
3. If you click OK then it prompt to enter "NAME" and/or "HOST" Once you edit any of those, previous one is deleted and new datasource is added.

File: ui/src/Datasources.tsx [hunk 1]
```diff
@@ -219,6 +219,76 @@ export const DatasourceForm = () => {
...
+            alert("Aborting as neither NAME or HOST URL was edited");
+          } else {
+            // set body 
+            body.name=nameds;
+            body.host=urlds;
+            body.type=ds.type;
+            body.headerAPIKeyName=ds.headerAPIKeyName;
+            body.headerAPIKeyValue=ds.headerAPIKeyValue;
+            try {
+              await axios.post(`${LOCAL_HOST}/datasource`, body, {
```
Comment (woodyjon) [inline ui/src/Datasources.tsx:268]: There is already a route to update datasources. Please do not post a new one and delete the old one. That would mean the did would change.

File: ui/src/Datasources.tsx [hunk 2]
```diff
@@ -245,6 +316,10 @@ export const DatasourcesList = () => {
                     <TableCell>{datasource.host}</TableCell>
                     <TableCell>{dayjs(datasource.CreatedAt).format('YYYY-MM-DD')}</TableCell>
                     <TableCell component="th" scope="row">{datasource.did}</TableCell>
+                    <TableCell>
+                      <Button variant="contained" onClick={e => handleEdit(datasource)}>Edit</Button>&nbsp;&nbsp;&nbsp;
```
Comment (woodyjon) [inline ui/src/Datasources.tsx:320]: Please use margins or paddings, instead of &nbsp.

**Review by woodyjon — CHANGES_REQUESTED** [review]

File: ui/src/Datasources.tsx (hunk 1 above)
Comment (ManishChain) [reply ui/src/Datasources.tsx:268]: Ok Jon. Thanks for update and I will revert the changes and try to use the existing route.

File: ui/src/Datasources.tsx (hunk 2 above)
Comment (ManishChain) [reply ui/src/Datasources.tsx]: Done

**Comment by ManishChain** [comment]
@goosecoid @woodyjon The test case for update feature testing is TestUpdateDatasourceByDID(). But the issue is that I am not able to retrieve the same datasource again ..see function TestUpdateDatasourceByDID_check()

**Review by goosecoid — APPROVED** [review]

**Pull request merged**